### ISSUE TITLE
[Request] Run custom script before running the program

### ISSUE THREAD

**Issue by Knebergish** [bottlesdevs/Bottles#948]
**Is your feature request related to a problem? Please describe.**
For the correct launch and work of the game _League of Legends_ before launching it is necessary to run a custom script ([script1 and script2](https://lutris.net/games/install/23547/view)). Lutris already has this feature, and it is used to run this game. Bottles can't do this yet, and the script needs to be run manually before each launch of the game. It would be very convenient to have the ability to automatically run the script in Bottles.

**Describe the solution you'd like**
Bottles already has the ability to customize startup options for specific programs. The ability to specify a custom script can be added in the same settings window.

**Describe alternatives you've considered**
An alternative is to run the script manually before launching the program. This works, but is not very convenient.

**Comment by mirkobrombin** [comment]
The `script2` seems unused

**Comment by Knebergish** [comment]
script2 is called inside script1. I have already checked their work on my machine, everything works fine,
This feature request is absolutely not critical, the usual quality of life.

**Comment by mirkobrombin** [comment]
I am not discussing whether or not to implement the such feature

**Comment by Knebergish** [comment]
Actually, when creating this issue, I had in mind something simpler: 
1) In the program launch settings window, a field for specifying the path to the script is added.
2) Before launching the program, this script is called in the background.

It is the responsibility of the user to create/download and store the script. If the script is located outside the sandbox, the user will have to give additional permission via Flatseal.
If the user needs to run several scripts, he can create one meta-script which will run all the others.

Your version, of course, is more comprehensive and means integration with the installers mechanism, but it means a very large amount of work. Perhaps this issue can be split into two:
1) The possibility of specifying pre- and post- scripts for the specific program manually.
2) Automatic download and configuration of installer scripts.

> Should users be able to simply enter a command instead of a full script?

I don't see reason of allowing user to specify one command instead of specifying a script file. This command can simply be placed in the file, so why complicate the UI?

**Comment by Knebergish** [comment]
> Another solution can be to change the input field to a FileChooser

I wrote "field", but meant exactly FileChooser; of course, that's the best option.

> Btw I'm fine with splitting this into 2 issues.

I think we can only consider the first part in this issue, and create another issue for the last part when necessary.

**Comment by mirkobrombin** [comment]
Implemented

**Comment by Lucki** [comment]
> Implemented

May I ask where this can be found?
I've found an option in the "launch options" to start a script *after* the program exited (https://docs.usebottles.com/bottles/programs) - but I can't find the way to set this for running a script beforehand.

**Comment by mirkobrombin** [comment]
This feature ended as a "before run" script.

**Comment by Lucki** [comment]
"ended" as in "wasn't implemented"?

I just verified that the "post-run"-script is indeed called after the program exits. So the description and translation is correct for this.

However, I can't find a "pre-run"-script option. Guess I'll have to resort to something like:

~~~ sh
/path/to/pre-run.sh && %command%
~~~

---

While doing the above I've noticed:
* White spaces in the script path aren't handled
* Launch options and the post-run-script setting will get ignored until Bottles is restarted
* Re-viewing the launch options won't show the current values until Bottles is restarted (probably related to the above)
* There's currently no way to see if there's currently a post-run script set (and which it is)
* The description for the launch options are translated which results telling me to put in `%befehl%` instead of `%command%` - that's wrong
